### PR TITLE
feat(sdk-coin-bch): refactor bch to its own module

### PR DESCRIPTION
**Pull request by ducnguyen-bitgo**
- Move bch, tbch and their tests to its own module.

Ticket: BG-51031

### CODE REVIEW

**Review by OttoAllmendinger — CHANGES_REQUESTED** [review]
we want to run the full test suite for all utxo coins (see discussion in `sdk-coin-btc`)

**Review by bitgoAaron — DISMISSED** [review]

**Comment by ducnguyen-bitgo** [comment]
@bitgoAaron I guess once this is approved and merged, I need to rebase the zec refactor again to avoid merge conflicts.

**Review by bitgoAaron — DISMISSED** [review]

**Review by mmcshinsky-bitgo — DISMISSED** [review]

File: modules/sdk-coin-bch/README.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+function init() {
+  const bitgoapi = new BitGoAPI({ env: 'test' });
+  bitgoapi.register('bch', Bch.createInstance);
+  bitgoapi.register('tbch', Tbch.createInstance);
+  return bitgoapi;
+}
+
+async function main() {
+  const bitgoapi = init();
+  const bch = bitgoapi.coin('bch');
```
Comment (mmcshinsky-bitgo) [inline modules/sdk-coin-bch/README.md:30]: We'll need `await` for these examples.

**Review by mmcshinsky-bitgo — DISMISSED** [review]

File: modules/sdk-coin-bch/test/unit/bch.ts [hunk 2]
```diff
@@ -1,12 +1,19 @@
...
 describe('Custom BCH Tests', function () {
-  const bch = getUtxoCoin('bch');
-  const tbch = getUtxoCoin('tbch');
+  const bitgo: TestBitGoAPI = TestBitGo.decorate(BitGoAPI, { env: 'test' });
+  bitgo.initializeTestVars();
+  bitgo.safeRegister('bch', Bch.createInstance);
+  bitgo.safeRegister('tbch', Tbch.createInstance);
+
+  const bch = bitgo.coin('bch') as Bch;
+  const tbch = bitgo.coin('tbch') as Tbch;
```
Comment (OttoAllmendinger) [inline modules/sdk-coin-bch/test/unit/bch.ts:16]: how about moving the setup boilerplate to `getUtxoCoin()`?

**Review by OttoAllmendinger — CHANGES_REQUESTED** [review]

File: modules/sdk-coin-bch/test/unit/bch.ts (hunk 2 above)
Comment (mmcshinsky-bitgo) [reply modules/sdk-coin-bch/test/unit/bch.ts:16]: Same comment as in `ltc`.

**Review by mmcshinsky-bitgo — APPROVED** [review]

**Review by OttoAllmendinger — APPROVED** [review]

**Pull request merged**